kuka_rsi_hw_interface/src/rsi_command_handler.cpp

```diff
@@ -56,17 +43,34 @@ RSICommandHandler::RSICommandHandler()
   RSol_el.Params()["A"] = xml::XMLParam(xml::XMLType::DOUBLE);
   RSol_el.Params()["B"] = xml::XMLParam(xml::XMLType::DOUBLE);
   RSol_el.Params()["C"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  xml::XMLElement AIPos_el("AIPos");
+  AIPos_el.Params()["A1"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  AIPos_el.Params()["A2"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  AIPos_el.Params()["A3"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  AIPos_el.Params()["A4"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  AIPos_el.Params()["A5"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  AIPos_el.Params()["A6"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  xml::XMLElement ASPos_el("ASPos");
+  ASPos_el.Params()["A1"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  ASPos_el.Params()["A2"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  ASPos_el.Params()["A3"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  ASPos_el.Params()["A4"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  ASPos_el.Params()["A5"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  ASPos_el.Params()["A6"] = xml::XMLParam(xml::XMLType::DOUBLE);
+  xml::XMLElement Delay_el("Delay");
+  Delay_el.Params()["D"] = xml::XMLParam(xml::XMLType::LONG);
   xml::XMLElement Ipoc_state_el("IPOC");
   Ipoc_state_el.Params()["IPOC"] = xml::XMLParam(xml::XMLType::LONG);
-  state_data_structure_.Childs().emplace_back(AIPos_el);
-  state_data_structure_.Childs().emplace_back(ASPos_el);
   state_data_structure_.Childs().emplace_back(RIst_el);
   state_data_structure_.Childs().emplace_back(RSol_el);
+  state_data_structure_.Childs().emplace_back(AIPos_el);
+  state_data_structure_.Childs().emplace_back(ASPos_el);
+  state_data_structure_.Childs().emplace_back(Delay_el);
   state_data_structure_.Childs().emplace_back(Ipoc_state_el);
 
   // Command structure
   command_data_structure_.Params()["Type"] = xml::XMLParam(xml::XMLType::STRING);
-  command_data_structure_.SetParam<xml::XMLString>("Type", xml::XMLString("KROSHU"));
+  command_data_structure_.SetParam<xml::XMLString>("Type", xml::XMLString("ImFree"));
```

Comment:
why roll back to ImFree?

Reply:
I did not wanted to commit it, but I did not wanted to bother how the simulator is working